Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/audit/COREX12RealtimeTransforms.java

```diff
@@ -0,0 +1,342 @@
+
+            if (areRequiredRequestFieldsNull(msg)) {
+                LOG.error("One or more fields in CORE X12 Realtime Request object were null.");
+                return true;
+            }
+
+
+        } catch (Exception exp) {
+            LOG.error("Encountered Error: " + exp.getMessage());
+            return true;
+        }
+        return false;
+    }
+
+    private boolean areRequiredCOREX12RealtimeFiledsNull(COREEnvelopeRealTimeRequest msg, AssertionType assertion) {
+        try {
+
+            if (assertion == null) {
+                LOG.error("Assertion object is null");
+                return true;
+            }
+            if (msg == null) {
+                LOG.error("COREEnvelopeRealTimeRequest object is null");
+                return true;
+            }
+
+            if (oXDR.areRequiredUserTypeFieldsNull(assertion)) {
+                LOG.error("One of more UserInfo fields from the Assertion object were null.");
+                return true;
+            }
+
+            if (areRequiredRequestFieldsNull(msg)) {
+                LOG.error("One or more fields in CORE X12 Realtime Request object were null.");
+                return true;
+            }
+
+        } catch (Exception exp) {
+            LOG.error("Encountered Error: " + exp.getMessage());
+            return true;
+        }
+        return false;
+    }
+
+    private boolean areRequiredRequestFieldsNull(COREEnvelopeRealTimeRequest msg) {
+        if (msg.getPayloadType() == null || msg.getPayloadType().length() <= 0) {
+            LOG.error("CORE X12 Realtime PayloadType is empty...");
+            return true;
+        }
+
+        if (msg.getProcessingMode() == null || msg.getProcessingMode().length() <= 0) {
+            LOG.error("CORE X12 Realtime ProcessingMode is empty...");
+            return true;
+        }
+
+        if (msg.getPayloadID() == null || msg.getPayloadID().length() <= 0) {
+            LOG.error("CORE X12 Realtime PayloadID is empty...");
+            return true;
+        }
+
+        if (msg.getTimeStamp() == null || msg.getTimeStamp().length() <= 0) {
+            LOG.error("CORE X12 Realtime TimeStamp is empty...");
+            return true;
+        }
+
+        if (msg.getSenderID() == null || msg.getSenderID().length() <= 0) {
+            LOG.error("CORE X12 Realtime SenderID is empty...");
+            return true;
+        }
+
+        if (msg.getReceiverID() == null || msg.getReceiverID().length() <= 0) {
+            LOG.error("CORE X12 Realtime ReceiverID is empty...");
+            return true;
+        }
+
+        if (msg.getCORERuleVersion() == null || msg.getCORERuleVersion().length() <= 0) {
+            LOG.error("CORE X12 Realtime CORERuleVersion is empty...");
+            return true;
+        }
+        return false;
+    }
+
+    private boolean areRequiredRequestFieldsNull(COREEnvelopeRealTimeResponse msg) {
+        if (msg.getPayloadType() == null || msg.getPayloadType().length() <= 0) {
```

Comment:
Are these all strings.  We have an existing null checker.  NullChecker.isNullish(msg.getPayloadType()).  A better check for string is: string == null || string.isEmpty(), which is what null checker does.